ast/ddl.go

```diff
@@ -672,6 +680,22 @@ func (n *ColumnDef) Accept(v Visitor) (Node, bool) {
 	return v.Leave(n)
 }
 
+func (n *ColumnDef) Validate() bool {
+	generatedCol := false
+	illegalOpt4gc := false
+	for _, opt := range n.Options {
+		if opt.Tp == ColumnOptionGenerated {
+			generatedCol = true
+		}
+		_, found := invalidOption4GeneratedColumn[opt.Tp]
+		illegalOpt4gc = illegalOpt4gc || found
+		if illegalOpt4gc {
+			break
```

Comment:
Remove the `break`. The program flow is wrong if those "invalid options" appeared before GENERATED.

```sql
create table t1 (a int generated always as (10) default 5);
create table t2 (a int default 5 generated always as (10));
```

The latter is currently *accepted* by TiDB, because we don't distinguish GENERATED options. Alternatively we could separate the column definitions as [MySQL described](https://dev.mysql.com/doc/refman/8.0/en/create-table.html).

```
column_definition:
    data_type [NOT NULL | NULL] [DEFAULT {literal | (expr)} ]
      [AUTO_INCREMENT] [UNIQUE [KEY]] [[PRIMARY] KEY]
      [COMMENT 'string']
      [COLLATE collation_name]
      [COLUMN_FORMAT {FIXED|DYNAMIC|DEFAULT}]
      [STORAGE {DISK|MEMORY}]
      [reference_definition]
      [check_constraint_definition]
  | data_type
      [COLLATE collation_name]
      [GENERATED ALWAYS] AS (expr)
      [VIRTUAL | STORED] [NOT NULL | NULL]
      [UNIQUE [KEY]] [[PRIMARY] KEY]
      [COMMENT 'string']
      [reference_definition]
      [check_constraint_definition]
```

Reply:
OK. I'll remove the break first and separate the column definitions later. maybe in another pr